core/vm/contracts.go

```diff
@@ -1353,3 +1358,91 @@ func (c *p256Verify) Run(input []byte) ([]byte, error) {
 		return nil, nil
 	}
 }
+
+// gpgEd25519Verify implements native verification for ed25519 signatures produced via gpg
+type gpgEd25519Verify struct{}
+
+var (
+	errMessageTooShort    = errors.New("message too short")
+	errPubKeyTooShort     = errors.New("public key too short")
+	errSignatureTooShort  = errors.New("signature too short")
+	errInvalidPublicKey   = errors.New("invalid public key format")
+	errInvalidSignature   = errors.New("invalid signature format")
+	errVerificationFailed = errors.New("signature verification failed")
+)
+
+// RequiredGas returns the gas required to execute the pre-compiled contract
+func (c *gpgEd25519Verify) RequiredGas(input []byte) uint64 {
+	// You can adjust this value based on your needs
+	return params.GpgEd25519VerifyGas
+}
+
+// Run performs ed25519 signature verification
+func (c *gpgEd25519Verify) Run(input []byte) ([]byte, error) {
+	// Input should be: message_len (32 bytes) || message || pubkey_len (32 bytes) || pubkey || sig_len (32 bytes) || signature
+	if len(input) < 96 { // minimum length for the three length fields
+		return nil, errMessageTooShort
+	}
+
+	// Extract message length and message
+	msgLen := new(big.Int).SetBytes(input[:32]).Uint64()
+	if len(input) < 32+int(msgLen) {
+		return nil, errMessageTooShort
+	}
+	message := input[32 : 32+msgLen]
+
+	// Extract public key length and public key
+	offset := 32 + msgLen
+	pubKeyLen := new(big.Int).SetBytes(input[offset : offset+32]).Uint64()
+	if len(input) < int(offset+32+pubKeyLen) {
+		return nil, errPubKeyTooShort
+	}
+	pubKey := input[offset+32 : offset+32+pubKeyLen]
+
+	// Extract signature length and signature
+	offset = offset + 32 + pubKeyLen
+	sigLen := new(big.Int).SetBytes(input[offset : offset+32]).Uint64()
+	if len(input) < int(offset+32+sigLen) {
```

Comment:
Is there a specific length the signature is supposed to be that we should verify?

Reply:
Via the command:
```bash
echo [message to sign] | xxd -r -p | gpg --pinentry-mode loopback --detach-sign --armor
```

Signing these two hex messages of varying lengths results in signatures of varying (byte) lengths:

1. "hellohello...."
```
68656c6c6f68656c6c6f68656c6c6f68656c6c6f68656c6c6f68656c6c6f68656c6c6f68656c6c6f68656c6c6f68656c6c6f68656c6c6f68656c6c6f68656c6c6f68656c6c6f68656c6c6f68656c6c6f68656c6c6f68656c6c6f68656c6c6f68656c6c6f68656c6c6f68656c6c6f68656c6c6f68656c6c6f68656c6c6f68656c6c6f68656c6c6f68656c6c6f68656c6c6f68656c6c6f68656c6c6f68656c6c6f68656c6c6f68656c6c6f68656c6c6f68656c6c6f68656c6c6f68656c6c6f68656c6c6f68656c6c6f68656c6c6f68656c6c6f68656c6c6f68656c6c6f68656c6c6f68656c6c6f68656c6c6f
```

2. "Hello, World"
```
48656c6c6f2c20576f726c64
```

Even though EdDSA signatures are deterministic length (64 bytes), the signature in armored format returned by gpg signing includes the actual signature plus various metadata. This armored signature length is non-deterministic (from what I have tested and researched).

Reply:
Agreed, no way to do this, will mark this as resolved.